feat(settings): Update 2FA Settings Row with backup codes subrow

vpomerleau opened the pull request:
## Because

* We are preparing to add a new recovery method for 2FA and are updating some design pieces

## This pull request

* Update the 2FA settings row design
* Add new SubRowBackupMethods component
* Updates stories, tests, l10n
* Add container-queries tailwind plugin to reflow content based on container size (not screen size)

## Issue that this pull request solves

Closes: #FXA-10206

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

[No codes available storybook](https://storage.googleapis.com/mozilla-storybooks-fxa/commits/0087d7cc268d34d9f331fafca2cea89ee1211784/fxa-settings/index.html?path=/story/components-settings-unitrowtwostepauth--tfa-enabled-no-codes-remaining)
![image](https://github.com/user-attachments/assets/2c49383d-6731-42a1-8b5f-14205947bd94)

[With codes remaining storybook](https://storage.googleapis.com/mozilla-storybooks-fxa/commits/0087d7cc268d34d9f331fafca2cea89ee1211784/fxa-settings/index.html?path=/story/components-settings-unitrowtwostepauth--tfa-enabled-with-codes-remaining)
![image](https://github.com/user-attachments/assets/de61c43d-885b-4d1a-82c0-902c5958129f)
[
Full settings page storybook](https://storage.googleapis.com/mozilla-storybooks-fxa/commits/0087d7cc268d34d9f331fafca2cea89ee1211784/fxa-settings/index.html?path=/story/pages-settings--completely-filled-out)

## Other information (Optional)

UX review ticket: https://mozilla-hub.atlassian.net/browse/FXA-10699

vbudhram reviewed (approved):
@vpomerleau Code looks good and works well locally. 

One minor UX confusion is that if a user clicks "Get new codes", a modal is popped up that says that operation can not be undone. However, this is only the case if the user confirms the codes on the next screen. If a user clicks cancel or "<-", codes would not have changed.

Wonder if we should update this to show two dialogs, "Are you sure you want to cancel making new codes?" if user clicks cancel. Then show modal when user confirms the new codes, that this can not be undone.

LZoog commented on packages/fxa-react/styles/ctas.css:
```diff
@@ -91,13 +91,13 @@
 
   @screen mobileLandscape {
-    .cta-base {
+    .cta-base:not(.cta-unit-row) {
```
If the sole purpose of having this class is so that we can use it in a `:not` here, can we instead create a `.cta-base-common` class or whatever name (shared maybe?), and then extend that for `cta-base`, but directly use the class instead of `cta-unit-row`? It would help us avoid needing this specificity at least and should be a little more maintainable.

LZoog commented on packages/fxa-settings/src/components/Icons/index.tsx:
```diff
@@ -71,8 +93,8 @@ export const BackupCodesDisabledIcon = ({
     Image={BackupCodesDisabled}
-    ariaLabel="Backup codes disabled"
-    ariaLabelFtlId="backup-codes-disabled-icon-aria-label"
+    ariaLabel="Backup authentication codes disabled"
```
👍

LZoog commented on packages/fxa-settings/src/components/Settings/SettingsSubRow/index.stories.tsx:
```diff
@@ -0,0 +1,72 @@
+import React from 'react';
```
Nit: missing license

Just wondering, why `SettingsSubRow`? I think it's fine but we have `UnitRow` so eh, could go just `SubRow`?

LZoog commented on packages/fxa-settings/src/components/Settings/SettingsSubRow/index.stories.tsx:
```diff
@@ -0,0 +1,72 @@
+      return (
+        // @container/unitRow on the container div allows sub row to adjust based on the size of the parent container
+        // instead of the viewport. This fixes issues with the subrow and CTAs overflowing their parent container in mobileLandscape
+        // and tablet portrait views.
```
Nice comment! TIL.

LZoog commented on packages/fxa-settings/src/components/Settings/SettingsSubRow/index.tsx:
```diff
@@ -0,0 +1,140 @@
+  showDescription,
+  title,
+}: SettingsSubrowProps) => {
+  const StatusIcon = () =>
```
Just a nit but to keep the span classes consistent, could do something like this?

```
  <span className="grow-0 shrink-0">
    {isEnabled ? (
      <CheckmarkGreenIcon className="mt-1" mode="enabled" />
    ) : (
      <AlertIcon className="m-0" mode="attention" />
    )}
  </span>
```

LZoog commented on packages/fxa-react/configs/tailwind.js:
```diff
@@ -107,6 +107,12 @@ module.exports = {
       desktop: '1024px',
       desktopXl: '1441px',
     },
+    containers: {
```
Since this is just JS I guess we could pull the object above into a constant and then spread it there and here to keep these two things consistent, but eh.

LZoog commented on packages/fxa-settings/src/components/Settings/UnitRowTwoStepAuth/index.tsx:
```diff
@@ -21,213 +21,239 @@ export const UnitRowTwoStepAuth = () => {
+            id="tfa-row-disable-modal-explain-1"
+            elems={{
+              linkExternal: (
+                <LinkExternal
```
Just another nit, you could set `LinkExternal` here to a variable and then you can reference it here and in the fallback, just to keep them consistent. (we do this in `CardHeader` with `spanElement`)

LZoog commented on packages/fxa-settings/src/components/Settings/UnitRowTwoStepAuth/index.tsx:
```diff
@@ -21,213 +21,239 @@ export const UnitRowTwoStepAuth = () => {
+            }}
+          >
+            <p className="text-center">
+              You won't be able to undo this action. You also have the option of{' '}
```
This looks like a straight apostrophe and not a curly one 👀

LZoog commented on packages/fxa-settings/src/components/Settings/UnitRowTwoStepAuth/index.tsx:
```diff
@@ -21,213 +21,239 @@ export const UnitRowTwoStepAuth = () => {
+        subRows={getSubRows()}
+      >
+        {exists && verified ? (
+          <>
```
Is this fragment needed?

vpomerleau replied:
Nope, thanks!

LZoog reviewed (commented):
Mostly just a bunch of nits. Looks good in storybook, but didn't test locally. :) Great job here, this looks like it was complicated and hard to work through!

vpomerleau commented:
> @vpomerleau Code looks good and works well locally.

Thanks for the review and testing!!

> One minor UX confusion is that if a user clicks "Get new codes", a modal is popped up that says that operation can not be undone. However, this is only the case if the user confirms the codes on the next screen. If a user clicks cancel or "<-", codes would not have changed.

That's a very good point - I might even go a step further and suggest removing the requirement to confirm a code. Seems like hand holding that might actually cause more problems for our users if they replace their saved codes and don't click through to finish the flow. (then the new codes aren't valid and the user is actually left with _no_ valid codes. That's a bigger change though, so for now I removed the modal and changed the wording on the final screen where the old codes get invalidated.
